backend/src/modules/authentication/core/application/command/SetPasswordCommandHandler.ts

```diff
@@ -14,8 +15,8 @@ export class SetPasswordCommandHandler implements CommandHandler<SetPassword> {
   ) {}
 
   async execute(command: SetPassword): Promise<CommandResult> {
-    const userAccount = await this.repository.findByEmail(command.email);
-    const { state, events } = setPasswordForUserAccount(userAccount, command, this.currentTimeProvider());
+    const userAccount = await this.repository.findById(command.userId);
```

Comment:
Czy kazdy user moze kazdemu ustawic haslo? Tutaj sie przyda sprawdzenie wlasnie aktualnego usera. Ale to moze byc kolejny PR :)

Reply:
czyli, że w metodzie setPasswordForUserAccount by się przydało jeszcze sprawdzenie np. tokena? Jeśli tak to kumam o co biega, ale nie mam zielonego pojęcia jak to zrobić .... to by musiało iść przez ten middleware do sprawdzania poprawności tokena, a to trzeba osobno skodzić, Si?

Reply:
hmmm... no to zalezy :P Najprosciej mozna to zrobic w metodzie execute, ale niezbyt to reuzywalne. Middleware wydaje sie lepszy. Chyba byly na kursach jakies takie rzeczy, poszukaj jak zrobic autoryzacje poprzez middleware. Ta autoryzacja mialaby tak dzialac, ze tylko user moze zmienic haslo dla samego siebie.